From ce6c7ecd3f765c96a29577488015c0d937a9f997 Mon Sep 17 00:00:00 2001
From: Biju Chakkachattil <bijux.chakkachattil@intel.com>
Date: Mon, 22 Jun 2015 04:49:23 +0530
Subject: [PATCH] EM/Type-c: Revert "Update pmic reg for enabling host-mode"

This reverts commit 944c09f40a3b7ddf9c8a5a3013899bf0e1bbedbc.

This is the workaround to enable OTG. This is no more required
as the BIOS 4.82 has included the required changes.

Change-Id: Ifc16917b6695360d12f927956526d77533bd4641
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-36254
Signed-off-by: Biju Chakkachattil <bijux.chakkachattil@intel.com>
---
 drivers/usb/typec/usb_typec_detect.c | 9 ---------
 1 file changed, 9 deletions(-)

Index: linux/drivers/usb/typec/usb_typec_detect.c
===================================================================
--- linux.orig/drivers/usb/typec/usb_typec_detect.c	2015-06-26 17:13:58.993217708 -0700
+++ linux/drivers/usb/typec/usb_typec_detect.c	2015-06-26 17:14:53.065217080 -0700
@@ -264,11 +264,6 @@
 
 			atomic_notifier_call_chain(&detect->otg->notifier,
 				USB_EVENT_ID, NULL);
-			/* [WA] Since neccessary pmic-gpio isn't exposed
-			 * as of now, this is a w/a to write directly.
-			 */
-			intel_soc_pmic_writeb(0x6e2d, 0x31);
-
 
 			return;
 		} else if (CC_RA(cc1.v_rd) && CC_RA(cc2.v_rd)) {
@@ -516,10 +511,6 @@
 			typec_enable_autocrc(detect->phy, false);
 		} else {
 			/* state = DFP; disable VBUS */
-			/* [WA] Since neccessary pmic-gpio isn't exposed
-			 * as of now, this is a w/a to write directly.
-			 */
-			intel_soc_pmic_writeb(0x6e2d, 0x30);
 			if (detect->is_pd_capable)
 				extcon_set_cable_state(detect->edev,
 						TYPEC_CABLE_USB_DFP, false);
